src/client/Microsoft.Identity.Client/Http/IHttpManager.cs

```diff
@@ -14,6 +14,7 @@ namespace Microsoft.Identity.Client.Http
     internal interface IHttpManager
     {
         long LastRequestDurationInMs { get; }
+        HttpClientHandler HttpClientHandler { get; set; }
```

Comment:
See my comments below. We should not make the class mutable, it will be very difficult to reason with because there is static caching in here. 

How about adding a HttpClientHandler in the SendRequedstAsync instead? Or better yet, add the callback as an input param of that factory? HttpManger already uses HttpClientHandler for mtls purposes.